monai/data/image_writer.py

```diff
@@ -258,11 +260,18 @@ def resample_if_needed(
                 ``np.float64`` for best precision. If ``None``, use the data type of input data.
                 The output data type of this method is always ``np.float32``.
         """
+        orig_type = type(data_array)
+        data_array = convert_to_tensor(data_array, track_meta=True)
+        if affine is not None:
+            data_array.affine = convert_to_tensor(affine, track_meta=False)  # type: ignore
         resampler = SpatialResample(mode=mode, padding_mode=padding_mode, align_corners=align_corners, dtype=dtype)
-        output_array, target_affine = resampler(
-            data_array[None], src_affine=affine, dst_affine=target_affine, spatial_size=output_spatial_shape
-        )
-        return output_array[0], target_affine
+        output_array = resampler(data_array[None], dst_affine=target_affine, spatial_size=output_spatial_shape)
+        # convert back at the end
+        if isinstance(output_array, MetaTensor):
+            output_array.applied_operations = []
+        data_array, *_ = convert_data_type(output_array, output_type=orig_type)  # type: ignore
+        affine, *_ = convert_data_type(output_array.affine, output_type=orig_type)  # type: ignore
```

Comment:
May I know why convert it back? Because the nibabel or ITK only supports numpy array?

Thanks.

Reply:
yes, that's the reason, for compatibility.